criu/uffd.c

```diff
@@ -1485,6 +1485,5 @@ int cr_lazy_pages(bool daemon)
 
 	tls_terminate_session();
 
-	xfree(events);
```

Comment:
could you explain in the commit message why it is wrongly added?

I see it is allocated in epoll_prepare, but I don't see where we release it.

Reply:
It's complicated. In certain error cases `events` could have been already free'd in xrealloc (, but it is very unlikely. It does not seem to be able to handle this at this level for error cases correctly. I will just drop this one.

Re-reading the man-page of `realloc` this patch might indeed be wrong, because it says, that if it fails it will not change `events`. So it can be dropped.

Reply:
```
diff --git a/criu/util.c b/criu/util.c
index 670ef158d..2aeb53b43 100644
--- a/criu/util.c
+++ b/criu/util.c
@@ -1312,6 +1312,7 @@ int epoll_prepare(int nr_fds, struct epoll_event **events)
 
 free_events:
        xfree(*events);
+       *events = NULL;
        return -1;
 }
 
```